### PR TITLE
Expand tests of resource resolution

### PR DESCRIPTION
**Pull request by bcsgh**
Add some more test for resource resolution.

This includes
- a test exposing a bug: https://github.com/etr/libhttpserver/issues/308
- some tests of some of the corners relating to another PR: https://github.com/etr/libhttpserver/pull/306

### CODE REVIEW

File: test/integ/basic.cpp [hunk 1]
```diff
@@ -358,6 +369,68 @@ LT_BEGIN_AUTO_TEST(basic_suite, two_endpoints)
...
+
+    // Check how family interacts.
+    LT_CHECK_EQ(false, ws->register_resource("OK", &ok2, true));
+
+    // Check that switched case does the right thing, whatever that is here.
+#ifdef CASE_INSENSITIVE
+    LT_CHECK_EQ(true, ws->register_resource("ok", &ok2));
+#else
+    LT_CHECK_EQ(false, ws->register_resource("ok", &ok2));
+#endif
```
Comment (bcsgh) [inline test/integ/basic.cpp:390]: After a bit more digging, I think these are actually backwards. And both in fact end up returning false.

tl;dr: `http_endpoint::operator <` is unconditionally case-insensitive:
https://github.com/etr/libhttpserver/blob/6d1df063501c9a7b75d68958fa3105a154a75cf0/src/details/http_endpoint.cpp#L144-L146 

That's going to be a pain to fix as it's something users could be inadvertently depending on.
